prov/usnic/src/usdf_dgram.c

```diff
@@ -128,33 +128,35 @@ ssize_t
 usdf_dgram_recvmsg(struct fid_ep *fep, const struct fi_msg *msg, uint64_t flags)
 {
 	return usdf_dgram_recvv(fep, msg->msg_iov, msg->desc,
-		msg->iov_count, (fi_addr_t)msg->addr, msg->context);
-}
-
-static inline ssize_t
-_usdf_dgram_send(struct usdf_ep *ep, struct usdf_dest *dest,
-		const void *buf, size_t len,  void *context)
-{
-	if (len <= USD_SEND_MAX_COPY - sizeof(struct usd_udp_hdr)) {
-		return usd_post_send_one_copy(ep->e.dg.ep_qp,
-			&dest->ds_dest, buf, len, USD_SF_SIGNAL, context);
-	} else {
-		return usd_post_send_one(ep->e.dg.ep_qp, &dest->ds_dest,
-			buf, len, USD_SF_SIGNAL, context);
-	}
+		msg->iov_count, (fi_addr_t) msg->addr, msg->context);
 }
 
 ssize_t
 usdf_dgram_send(struct fid_ep *fep, const void *buf, size_t len, void *desc,
 		fi_addr_t dest_addr, void *context)
 {
-	struct usdf_ep *ep;
 	struct usdf_dest *dest;
+	struct usdf_ep *ep;
+	uint8_t completion = 0;
 
 	ep = ep_ftou(fep);
-
 	dest = (struct usdf_dest *)(uintptr_t) dest_addr;
-	return _usdf_dgram_send(ep, dest, buf, len, context);
+
+	if (ep->ep_tx_dflt_signal_comp ||
+		(ep->e.dg.tx_op_flags & FI_COMPLETION))
+		completion = USD_SF_SIGNAL;
+
+	if (len + sizeof(struct usd_udp_hdr) <= USD_SEND_MAX_COPY) {
+		return usd_post_send_one_copy(ep->e.dg.ep_qp, &dest->ds_dest,
+					      buf, len, completion, context);
```

Comment:
Looks like spaces are used to line the indent up nicely.  While I agree this looks nicer, it's not the style we're trying to match here.  Tabs only for indents at the beginning of lines, please.  Just replace these 6 spaces with a hard tab.